File: ruby/sum-of-multiples/sum_of_multiples.rb
class SumOfMultiples
  attr_reader :multiples

  def initialize(*multiples)
    @multiples = multiples
  end

  def to(n)
    (1...n).filter do |x|
      multiples.any? { |m| x % m == 0 }
    end.sum
  end
end

File: kotlin/sum-of-multiples/src/main/kotlin/SumOfMultiples.kt
object SumOfMultiples {
    fun sum(multiplesOf: Set<Int>, upTo: Int): Int
        = (1 until upTo).filter { multiplesOf.multipleOfAny(it) }.sum()

    private fun Iterable<Int>.multipleOfAny(num: Int): Boolean = this.any { num % it == 0 }
}
